src/modules/RA/DataLayer/components/tabs/MinisheetTab/index.js

```diff
@@ -0,0 +1,119 @@
+import React, { useCallback, useMemo } from 'react';
+
+import {
+  TextInput,
+  BooleanInput,
+  useTranslate,
+} from 'react-admin';
+
+import { useField } from 'react-final-form';
+
+import { ColorInput } from 'react-admin-color-input';
+import { makeStyles } from '@material-ui/core/styles';
+import Typography from '@material-ui/core/Typography';
+import Card from '@material-ui/core/Card';
+import CardContent from '@material-ui/core/CardContent';
+
+import MiniSheetFieldTree from './MiniSheetFieldTree';
+
+import HelpContent from '../../../../../../components/react-admin/HelpContent';
+import FieldGroup from '../../../../../../components/react-admin/FieldGroup';
+
+const useStyles = makeStyles({
+  colorPicker: {
+    width: '25%',
+  },
+  addPopup: {
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    height: '40vh',
+  },
+  card: {
+    display: 'flex',
+    flexDirection: 'row',
+  },
+  title: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    width: '100%',
+    borderBottom: 'thin solid grey',
+    marginBottom: '25px',
+  },
+});
+
+
+const MinisheetTab = () => {
+  const classes = useStyles();
+  const translate = useTranslate();
+  const { input: { value: fields } } = useField('fields');
+  const {
+    input: {
+      value: {
+        advanced,
+        enable,
+        wizard: { sections = [], title } = {},
+      },
+    },
+  } = useField('minisheet_config');
+
+
+  const getAvailableFields = useCallback(() => {
+    const sectionsFieldIds = sections.reduce((fieldIds, { children = [] }) => {
+      const sectionFieldIds = children.map(({ sourceFieldId }) => sourceFieldId);
+      return [...fieldIds, ...sectionFieldIds];
+    }, []);
```

Comment:
To be tested but i think you can simplify this const by using `flatMap`

=> `const sectionsFieldIds = sections.flatMap(({ children = [] }) => children.map(({ sourceFieldId }) => sourceFieldId))`

Reply:
This is nice, I'll try it!

Reply:
Done!